### PR TITLE
RES-1006

### PR DESCRIPTION
**Pull request by mvacaporale**
Updates to DSConv2d layer including bug fixes and merging with previous directory refactoring.

### CODE REVIEW

File: nupic/research/frameworks/dynamic_sparse/networks/layers.py [hunk 1]
```diff
@@ -438,43 +490,137 @@ def update_coactivations(self, input_tensor, output_tensor):
...
+                h = torch.sum(corr_proxy, (0, 2, 3))
+                h = h.type(self.coactivations.dtype)
+
+                del corr_proxy
 
             self.coactivations[self.connection_indxs] += h
 
             del h
 
+            # grouped_input = input_tensor.repeat((1, self.new_groups, 1, 1))
```
Comment (subutai) [inline nupic/research/frameworks/dynamic_sparse/networks/layers.py]: Do we really need to keep this commented out code? If not, best to remove it.

File: nupic/research/frameworks/dynamic_sparse/networks/layers.py [hunk 2]
```diff
@@ -639,57 +766,123 @@ def progress_connections(self):
...
+                self.c01_frac = calc_onfrac(self.last_c01_mask)
+                self.c01_frac_rel = calc_onfrac(self.last_c01_mask[~last_keep_mask])
+                self.c11_num = float(self.last_c11_mask.sum())
+                self.c11_frac = calc_onfrac(self.last_c11_mask)
+                self.c11_frac_rel = calc_onfrac(self.last_c11_mask[last_keep_mask])
+                self.c00_num = float(self.last_c00_mask.sum())
+                self.c00_frac = calc_onfrac(self.last_c00_mask)
+                self.c00_frac_rel = calc_onfrac(self.last_c00_mask[~last_keep_mask])
+
+                # if self.survival_rate is not None:
```
Comment (subutai) [inline nupic/research/frameworks/dynamic_sparse/networks/layers.py]: Do we really need to keep this commented out code? If not, best to remove it.

File: nupic/research/frameworks/dynamic_sparse/networks/layers.py [hunk 3]
```diff
@@ -717,59 +910,84 @@ class RandDSConv2d(DSConv2d):
...
     """
 
-    def progress_connections(self, *args, **kwargs):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.coactivations.data[:] = torch.ones_like(self.weight)
 
-        if self.prune_grads_hook is not None:
-            self.prune_grads_hook.remove()
+    # def progress_connections(self, *args, **kwargs):
```
Comment (subutai) [inline nupic/research/frameworks/dynamic_sparse/networks/layers.py]: Do we really need to keep this commented out code? If not, best to remove it.

File: nupic/research/frameworks/dynamic_sparse/networks/layers.py [hunk 4]
```diff
@@ -717,59 +910,84 @@ class RandDSConv2d(DSConv2d):
...
         pass
 
+    def update_coactivations(self, *args, **kwargs):
+        pass
 
-class SparseConv2d(SparseWeights2d):
-    """
-    Conv layer with static sparsity.
-    """
+# class SparseConv2d(SparseWeights2d):
```
Comment (subutai) [inline nupic/research/frameworks/dynamic_sparse/networks/layers.py]: Do we really need to keep this commented out code? If not, best to remove it.

File: nupic/research/frameworks/dynamic_sparse/models/main.py [hunk 5]
```diff
@@ -80,7 +80,7 @@ def setup(self):
             )
 
         # add a learning rate scheduler
-        if self.lr_scheduler or self.lr_scheduler == "MultiStepLR":
+        if self.lr_scheduler == "MultiStepLR":
```
Comment (lucasosouza) [inline nupic/research/frameworks/dynamic_sparse/models/main.py]: This change is gonna impact some run files - I've seen it is called StepLR on some files. Are you sure on this?

File: nupic/research/frameworks/dynamic_sparse/networks/layers.py (hunk 4 above)
Comment (mvacaporale) [reply nupic/research/frameworks/dynamic_sparse/networks/layers.py]: Good point, I'll remove this and the others above.

File: nupic/research/frameworks/dynamic_sparse/models/main.py (hunk 5 above)
Comment (mvacaporale) [reply nupic/research/frameworks/dynamic_sparse/models/main.py]: 1) The following `elif` statement checks for `self.lr_scheduler == "StepLR"`.
2) Currently, there's a bug: Without removing `or self.lr_scheduler` you'd evaluate `if self.lr_scheduler` to `true` for any non-empty input, and so the scheduler would be invariably `"MultiStepLR"` no matter what you ask for.
